src/ui/public/agg_types/metrics/derivative.js

```diff
@@ -0,0 +1,142 @@
+import AggTypesMetricsMetricAggTypeProvider from 'ui/agg_types/metrics/metric_agg_type';
+import orderAggTemplate from 'ui/agg_types/controls/sub_agg.html';
+import _ from 'lodash';
+import $ from 'jquery';
+import VisAggConfigProvider from 'ui/vis/agg_config';
+import VisSchemasProvider from 'ui/vis/schemas';
+
+export default function AggTypeMetricDerivativeProvider(Private) {
+  const DerivativeAggType = Private(AggTypesMetricsMetricAggTypeProvider);
```

Comment:
I think this name is confusing - it's not a DerivativeAggType but a MetricAggType.  The way it is named now, it looks like a subclass of MetricAggType, but if that was the case, you wouldn't have to pass in all the options to specialize the metric for a derivative type.